packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,395 @@
+        [],
+        {
+          showOutput: false,
+          showPopup: true,
+          progress: overallProgress,
+        },
+      );
+      return;
+    }
+  } else if (action === InstallationAction.Install) {
+    let packageFullName = selectedEmitter.package;
+    if (version) {
+      packageFullName = `${selectedEmitter.package}@${version}`;
+    }
+    logger.info(`Installing ${packageFullName}`);
+    /* verify dependency packages. */
+    const dependenciesToInstall = await npmUtil.ensureNpmPackageDependencyInstall(
+      selectedEmitter.package,
+      version,
+      npmDependencyType.peerDependencies,
+    );
+    logger.info(`${dependenciesToInstall}`);
+    if (dependenciesToInstall.length > 0) {
+      vscode.window.showInformationMessage(
+        `Need to manually upgrade following dependency packages: ${dependenciesToInstall.join("\\n")}. \nGenerating Cancelled`,
+        "OK",
+      );
+      return;
+    }
+    packagesToInstall.push(`${packageFullName}`);
+  }
+
+  /* npm install packages. */
+  if (packagesToInstall.length > 0) {
+    logger.info(`Installing ${packagesToInstall.join("\n\n")} under ${baseDir}`, [], {
+      showOutput: true,
+      showPopup: true,
+      progress: overallProgress,
+    });
+    try {
+      const npmInstallResult = await npmUtil.npmInstallPackages(packagesToInstall, undefined, {
+        onStdioOut: toOutput,
+        onStdioError: toError,
+      });
+      if (npmInstallResult.exitCode !== 0) {
+        logger.error(`Error occurred when installing packages.`, [`${npmInstallResult.stderr}`], {
+          showOutput: true,
+          showPopup: true,
+          progress: overallProgress,
+        });
+        return;
+      }
+      logger.info("completed install...");
+    } catch (err) {
+      logger.error(`Exception occurred when installing packages.`, [err], {
+        showOutput: true,
+        showPopup: true,
+        progress: overallProgress,
+      });
+      return;
+    }
+  }
+
+  /* emit */
+  logger.info("Generating ...", [], {
+    showOutput: false,
+    showPopup: false,
+    progress: overallProgress,
+  });
+
+  const cli = await resolveTypeSpecCli(baseDir);
+  if (!cli) {
+    logger.error(
+      "Cannot find TypeSpec CLI. Please install @typespec/compiler. Generating Cancelled.",
+      [],
+      {
+        showOutput: true,
+        showPopup: true,
+        progress: overallProgress,
+      },
+    );
+    return;
+  }
+  const outputDir = path.resolve(baseDir, selectedEmitter.emitterKind, selectedEmitter.language);
+
+  const options: Record<string, string> = {};
+  options["emitter-output-dir"] = outputDir;
+  logger.info(
+    `Start to generate ${selectedEmitter.language} ${selectedEmitter.emitterKind} code under ${outputDir}...`,
+    [],
+    {
+      showOutput: true,
+      showPopup: false,
+      progress: overallProgress,
+    },
+  );
+  try {
+    const compileResult = await compile(cli, mainTspFile, selectedEmitter.package, options);
```

Comment:
i think you can use client.runCliCommand to trigger the compile command

Reply:
I use spawExecution. Is there any reason that we need to use client.runCliCommand?

Reply:
easier to use... you dont need to do the same work again, and it spawExecution too...

Reply:
we need to wrap the args, so I write a `compile` function to do so.